### PR TITLE
Scaffold flask application

### PR DESCRIPTION
**Pull request by lcarva**

### CODE REVIEW

**Comment by lcarva** [comment]
@mprahl PTAL

It's just a very simple layout for the application.

File: .gitignore [hunk 1]
```diff
@@ -103,5 +103,8 @@ venv.bak/
 # mypy
 .mypy_cache/
 
+# Flask
+settings.cfg
```
Comment (mprahl) [inline .gitignore]: Why is it `.cfg` and not `.py`?

File: cachito/__init__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+app = Flask(__name__)
+app.config.from_object('cachito.default_settings')
+app.config.from_envvar('CACHITO_SETTINGS')
+
+if not app.debug:
+    import logging
+    from logging.handlers import TimedRotatingFileHandler
+    # https://docs.python.org/3.6/library/logging.handlers.html#timedrotatingfilehandler
+    file_handler = TimedRotatingFileHandler(os.path.join(app.config['LOG_DIR'], 'cachito.log'),
```
Comment (mprahl) [inline cachito/__init__.py]: I don't think we want file logs. I think we want to stream them to stdout for OpenShift captures them.

File: cachito/__init__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import os
+from flask import Flask
+
+app = Flask(__name__)
```
Comment (mprahl) [inline cachito/__init__.py]: Let's refactor this to use an application factory. That can be in a separate PR.

File: cachito/views.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+from flask import render_template
+
+from cachito import app
+
+
+@app.route('/')
+def index():
+    app.logger.warning('sample message')
```
Comment (mprahl) [inline cachito/views.py]: This can be removed

File: tests/test_cachito.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import unittest
+
+import cachito
+
+
+class CachitoTestCase(unittest.TestCase):
```
Comment (mprahl) [inline tests/test_cachito.py]: I'd rather use pytest tests and pytest fixtures for the test client like we do in Greenwave and Estuary

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+[testenv:lint]
+deps =
+    flake8
+commands =
+    python -m flake8 {posargs}
+
+[flake8]
+show-source = True
+max-line-length = 100
+exclude = .git,.tox,dist,*egg,*fedmsg.d
```
Comment (mprahl) [inline tox.ini]: Oh no, not fedmsg!

File: tox.ini [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+deps =
+    -rdev-requirements.txt
+whitelist_externals =
+    make
+    touch
+setenv =
+	CACHITO_SETTINGS={toxinidir}/settings.cfg
+commands =
+    touch {toxinidir}/settings.cfg
+    py.test tests/ {posargs}
```
Comment (mprahl) [inline tox.ini]: It's nice to use `-vvv` here so we can get more information about failures

**Comment by mprahl** [comment]
I'd like to have the stuff I mentioned in the comments addressed but we can handle those in other PRs.

:+1:

File: cachito/__init__.py (hunk 2 above)
Comment (lcarva) [reply cachito/__init__.py]: Removed it. Part of the cookiecutter stuff. Dropped it for now.

File: cachito/__init__.py (hunk 3 above)
Comment (lcarva) [reply cachito/__init__.py]: Awesome, thanks for suggesting that. This is done now.

File: tox.ini (hunk 6 above)
Comment (lcarva) [reply tox.ini]: Honestly, I don't know how that got in there. Removed.

File: .gitignore (hunk 1 above)
Comment (lcarva) [reply .gitignore]: Changed it to .py.

Comment (lcarva) [reply .gitignore]: Silly cookiecutter...

**Comment by lcarva** [comment]
@mprahl, thanks for the review! This should be ready for another round.

File: .gitignore [hunk 8]
```diff
@@ -97,11 +97,17 @@ venv.bak/
 # Rope project settings
 .ropeproject
 
+# VS Code settings
+.vscode
```
Comment (mprahl) [inline .gitignore]: 👍
